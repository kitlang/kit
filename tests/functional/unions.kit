union MyUnion {
    public static var xyz = "hi";

    var a: Char;
    var b: CString;
}

function main() {
    printf("%s\n", MyUnion.xyz);
    var a: MyUnion;
    a.a = 100;
    printf("%hhi\n", a.a);
    a.b = "abcdefg";
    printf("%s\n", a.b);

    puts("MyUnion fields:");
    for field in MyUnion.fields {
        printf("- ");
        puts(field);
    }

    var x = union MyUnion {a: c'a'};
    var y = union MyUnion {b: "hello"};
}
