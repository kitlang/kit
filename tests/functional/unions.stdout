hi
100
abcdefg
MyUnion fields:
- a
- b
